apps/permissions.py

```diff
@@ -62,6 +62,32 @@ def has_owner_access(request, obj):
     return (organization_id and obj.owner_id == organization_id) or obj.owner_id == user_id
 
 
+def parse_wallet_ids(request):
+    response = settings.REQUESTS_SESSION.get(
+        f'{settings.PROFILES_URL}/api/v1/wallet?page_size=9999',
+        headers={'Authorization': 'JWT {}'.format(get_jwt_from_request(request))}
+    )
+    if not response.ok:
+        raise Custom500Error(detail='Invalid response from profiles', status_code=response.status_code)
+
+    return [data['id'] for data in response.json()['data']]
+
+
+def wallet_owner_filter(request):
+    wallet_ids = parse_wallet_ids(request)
+
+    return Q(carrier_wallet_id__in=wallet_ids) | Q(shipper_wallet_id__in=wallet_ids) | \
+        Q(moderator_wallet_id__in=wallet_ids)
+
+
+def nested_wallet_owner_filter(request):
```

Comment:
I think you could dry these two filters up by dynamically adding the `shipment__` prefix. Something like
```
**{f'{"shipment__" if nested else ""}carrier_wallet_id__in': wallet_ids}
```